src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairPDHMM.java

```diff
@@ -0,0 +1,399 @@
+                if (rangeForReadOverlapToDeterminedBases < 0 ||
+                    allele.getMaximumExtentOfSiteDeterminedAlleles().overlapsWithMargin(
+                        unclippedSpan, rangeForReadOverlapToDeterminedBases + 1)) {
+                    // Do nothing, keep the computed likelihood
+                } else {
+                    mLogLikelihoodArray[readIndex * alleleCount + a] = Double.NEGATIVE_INFINITY;
+                }
+                logLikelihoods.set(a, readIndex, mLogLikelihoodArray[readIndex * alleleCount + a]);
+                writeToResultsFileIfApplicable(inputScoreImputator, read, allele, mLogLikelihoodArray[readIndex * alleleCount + a]);
+            }
+            readIndex++;
+        }
+
+        if(doProfiling) {
+            postProcessingTime += (System.nanoTime() - startTime);
+        }
+
+
+    }
+
+    @Override
+    public void close() {
+        pairPDHmm.done(); // TODO evidently this doesn't have a close command?
+        if (doProfiling) {
+            logger.info("Time spent in setup for JNI call : " + (pairHMMSetupTime * 1e-9));
+            logger.info("Time spent in JNI call : " + (pairHMMTime * 1e-9));
+            logger.info("Time spent in post-processing : " + (postProcessingTime * 1e-9));
+            // logger.info("Total calls : " + totalCalls);
+        }
+        super.close();
+    }
+
+    protected void writeToResultsFileIfApplicable(final PairHMMInputScoreImputator inputScoreImputator, GATKRead read, PartiallyDeterminedHaplotype allele, double lk)
+    {
+        if(debugOutputStream == null) return;
+
+        if (inputScoreImputator == null || read == null || allele == null) {
+            logger.warn("Null input provided to writeToResultsFileIfApplicable");
+            return;
+        }
+
+        final PairHMMInputScoreImputation inputScoreImputation = inputScoreImputator.impute(read);
+        final byte[] readBases = read.getBases();
+        final byte[] readQuals = read.getBaseQualities();
+        final byte[] readInsQuals = inputScoreImputation.insOpenPenalties();
+        final byte[] readDelQuals = inputScoreImputation.delOpenPenalties();
+        final byte[] overallGCP = inputScoreImputation.gapContinuationPenalties();
+
+        final byte[] alleleBases = allele.getBases();
+        final byte[] allelePDBases = allele.getAlternateBases();
+
+        try {
+            debugOutputStream.write("\n" + new String(alleleBases) + "\t" + Arrays.toString(allelePDBases) + "\t" + new String(readBases) + "\t" + SAMUtils.phredToFastq(readQuals) + "\t" + SAMUtils.phredToFastq(readInsQuals) + "\t" + SAMUtils.phredToFastq(readDelQuals) + "\t" + SAMUtils.phredToFastq(overallGCP) + "\t" + String.format("%e",lk));
+        } catch (IOException e) {
+            throw new GATKException("Error writing to specified HMM results output stream", e);
+        }
+
+    }
+
+    private class GKLSubmissionBatchManager {
```

Comment:
This doesn't seem to be used.  Probably delete it or at the very least comment it out with explanations of why we need to keep it.

Reply:
If we delete we should check that there aren't other things that could now also be deleted.

Reply:
it was my halfway step for batch submission before the gkl handled the array unrolling internally... i'm inclined to leave it commented out for now since it was a nice thing... hmm...